### PR TITLE
Onboarding: move SelectItems to their own component

### PR DESCRIPTION
**Pull request by arcangelini**
#### Changes proposed in this Pull Request

When migrating IntentScreen to packages the SelectItems and SelectItemsAlt components were bundled together. This PR moves them to their own component within `packages/onboarding`.

#### Testing instructions

1. Pull branch and `yarn start`
2. Go to http://calypso.localhost:3000/start/ and navigate to the intent step
3. Verify everything looks correct and no errors
4. Check http://calypso.localhost:3000/stepper as well

### CODE REVIEW

File: packages/onboarding/src/select-items-alt/index.tsx [hunk 1]
```diff
@@ -2,32 +2,40 @@ import { Button } from '@automattic/components';
...
+
+export interface SelectItemAlt< T > {
+	show: boolean;
+	key: string;
+	description: TranslateResult;
+	value: T;
+	actionText: TranslateResult;
+	disable: boolean;
+	disableText: TranslateResult;
+}
```
Comment (escapemanuele) [inline packages/onboarding/src/select-items-alt/index.tsx:17]: Naming is better this way, but search for "SelectAltItem" in the codebase, you'll find a couple places where it still needs to be changed! (for example `client/signup/steps/intent/intents.tsx`)

**Review by escapemanuele — CHANGES_REQUESTED** [review]
Great work, in this way we quickly solved our error. Just a couple of things to change and after that, it's good to go!

File: packages/onboarding/src/select-items-alt/index.tsx (hunk 1 above)
Comment (arcangelini) [reply packages/onboarding/src/select-items-alt/index.tsx:17]: Oof, nice catch. Updated the two instances I saw.

**Review by escapemanuele — APPROVED** [review]
:shipit:

**Pull request merged**